pkg/migrations/op_create_table_test.go

```diff
@@ -761,6 +761,107 @@ func TestCreateTable(t *testing.T) {
 				}, testutils.UniqueViolationErrorCode)
 			},
 		},
+		{
+			name: "create table with foreign key constraint",
+			migrations: []migrations.Migration{
+				{
+					Name: "01_create_referenced_table",
+					Operations: migrations.Operations{
+						&migrations.OpCreateTable{
+							Name: "owners",
+							Columns: []migrations.Column{
+								{
+									Name: "id",
+									Type: "int",
+								},
+								{
+									Name: "name",
+									Type: "text",
+								},
+								{
+									Name: "city",
+									Type: "text",
+								},
+							},
+							Constraints: []migrations.Constraint{
+								{
+									Name:    "pk_owners",
+									Type:    migrations.ConstraintTypePrimaryKey,
+									Columns: []string{"id"},
+								},
+							},
+						},
+					},
+				},
+				{
+					Name: "02_create_referencing_table",
+					Operations: migrations.Operations{
+						&migrations.OpCreateTable{
+							Name: "pets",
+							Columns: []migrations.Column{
+								{
+									Name: "id",
+									Type: "int",
+								},
+								{
+									Name: "owner_id",
+									Type: "int",
+								},
+								{
+									Name: "name",
+									Type: "text",
+								},
+							},
+							Constraints: []migrations.Constraint{
+								{
+									Name:    "fk_owners",
+									Type:    migrations.ConstraintTypeForeignKey,
+									Columns: []string{"owner_id"},
+									References: &migrations.ConstraintReferences{
+										Table:    "owners",
+										Columns:  []string{"id"},
+										OnDelete: migrations.ForeignKeyReferenceOnDeleteCASCADE,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			afterStart: func(t *testing.T, db *sql.DB, schema string) {
+				MustInsert(t, db, schema, "01_create_referenced_table", "owners", map[string]string{
+					"id":   "1",
+					"name": "alice",
+					"city": "new york",
+				})
+			},
```

Comment:
Can we add more assertions here? We should be able to test the FK constraint is enforced here I think.